neqo-qpack/src/encoder.rs

```diff
@@ -66,13 +68,15 @@ impl QPackEncoder {
         }
         qdebug!(
             [self],
-            "Set max capacity to {} {}.",
+            "Set max capacity to new capacity:{} old:{} max_table_size={}.",
             cap,
-            self.table.capacity()
+            self.table.capacity(),
+            self.max_table_size,
         );
-        self.max_entries = (cap.to_f64().unwrap() / 32.0).floor().to_u64().unwrap();
-        // we also set our table to the max allowed. TODO we may not want to use max allowed.
-        self.change_capacity(cap)
+        let new_cap = std::cmp::min(self.max_table_size, cap);
+        self.max_entries = (new_cap.to_f64().unwrap() / 32.0).floor().to_u64().unwrap();
```

Comment:
Since you are here, this should be:

```suggestion
        self.max_entries = min(self.max_table_size, cap) / 32;
```

Simple division truncates in the way we want and we don't need to round-trip to a float.